akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala

```diff
@@ -158,7 +158,7 @@ private[http] final class HttpRequestParser(
       uriEnd + 1
     }
 
-    override def onBadProtocol(): Nothing = throw new ParsingException(HttpVersionNotSupported)
+    override def onBadProtocol(detail: String): Nothing = throw new ParsingException(HttpVersionNotSupported, "")
```

Comment:
Why is the detail ignored here?

Reply:
I am keeping the original behavior for this specific case because at least personally I was only interested in the `The server-side HTTP version is not supported`.

Generally speaking if the protocol is wrong its very easy to figure out the cause because there are only 3 protocols, but the `The server-side HTTP version is not supported` gets thrown generically in many places.

I have no objection to adding the detail here if requested.